src/posix/platform/infra_if.cpp

```diff
@@ -515,6 +524,82 @@ void InfraNetif::ReceiveIcmp6Message(void)
     }
 }
 
+void InfraNetif::HandleHostAddressResponse(union sigval sv)
+{
+    struct gaicb *   req = (struct gaicb *)sv.sival_ptr;
+    struct addrinfo *res = (struct addrinfo *)req->ar_result;
+
+    const uint8_t kMaxHostAddresses = 10;
+    otIp6Address  addresses[kMaxHostAddresses];
+    uint8_t       numAddresses = 0;
+
+    otLogInfoPlat("Handling host address response...");
+
+    for (struct addrinfo *rp = res; rp != NULL && numAddresses < kMaxHostAddresses; rp = rp->ai_next)
+    {
+        struct sockaddr_in6 *ip6Addr;
+
+        if (rp->ai_family != AF_INET6)
+        {
+            continue;
+        }
+
+        ip6Addr = reinterpret_cast<sockaddr_in6 *>(rp->ai_addr);
+        memcpy(&addresses[numAddresses].mFields.m8, &ip6Addr->sin6_addr.s6_addr, OT_IP6_ADDRESS_SIZE);
+        numAddresses++;
+    }
+
+    otPlatInfraIfReceiveHostAddress(gInstance, mInfraIfIndex, (char *)req->ar_name, addresses, numAddresses);
+
+    freeaddrinfo(res);
+    freeaddrinfo((struct addrinfo *)req->ar_request);
+    free(req);
+}
+
+otError InfraNetif::RequestHostAddressAsync(uint32_t aInfraIfIndex, const char *aHostName)
+{
+    otError          error = OT_ERROR_NONE;
+    struct addrinfo *hints;
+    struct gaicb *   reqs[1];
+    struct sigevent  sig;
+    int              status;
+
+    VerifyOrExit(aInfraIfIndex == mInfraIfIndex, error = OT_ERROR_DROP);
+
+    hints = (struct addrinfo *)malloc(sizeof(struct addrinfo));
+    VerifyOrExit(hints != nullptr, error = OT_ERROR_DROP);
```

Comment:
```suggestion
    VerifyOrExit(hints != nullptr, error = OT_ERROR_NO_BUFS);
```

nit: should use OT_ERROR_NO_BUFS